app/knight_class.py

```diff
@@ -0,0 +1,18 @@
+class Knight:
+    def __init__(self, dict_knight: dict) -> None:
+        self.name = dict_knight["name"]
+        self.hp = dict_knight["hp"]
+        self.power = dict_knight["power"] + dict_knight["weapon"]["power"]
+        self.protection = sum(
+            [armour["protection"] for armour in dict_knight["armour"]]
+        )
+        self.potion = dict_knight["potion"]
```

Comment:
u can make like this "self.potion = self.drink_the_potion(dict_knight["potion"]) - so u will write less code + u can remove 8 line in battle_func.py